src/crypto.ts

```diff
@@ -195,4 +200,41 @@ export default class Crypto {
       decodeBase64(formSecretKey)
     )
   }
+
+  /**
+   * Download and decrypt the given attachments.
+   * @param formSecretKey Secret key as a base-64 string
+   * @param decryptParams The params containing encrypted content and information.
+   * @returns The decrypted attachments if successful. Else, null will be returned.
+   * @throws {MissingPublicKeyError} if a public key is not provided when instantiating this class and is needed for verifying signed content.
+   */
+  downloadAndDecryptAttachments = async(formSecretKey: string, decryptParams: DecryptParams): Promise<DecryptedAttachments | null> => {
+    let decryptedRecords: DecryptedAttachments = {}
+
+    const attachmentRecords: EncryptedAttachmentRecords = decryptParams.attachmentDownloadUrls ?? {}
+    const decryptedContent = this.decrypt(formSecretKey, decryptParams)
+    if (decryptedContent === null) return null
+
+    for (let fieldId in attachmentRecords) {
+      const downloadResponse = await axios.get(attachmentRecords[fieldId], { responseType: 'json' })
+      if (downloadResponse.status !== 200) return null
+
+      let data = downloadResponse.data
+      data.encryptedFile.binary = decodeBase64ToUint8Array(data.encryptedFile.binary)
+      const decryptedFile = await this.decryptFile(formSecretKey, data.encryptedFile)
+
+      let filename = 'Unknown File.data'
+      for (const i in decryptedContent.responses) {
```

Comment:
rather than iterating through the responses for each attachment, might it be better to create a map of fieldId -> answer outside of the outer `for` loop? that way we only have to iterate through the attachments once.